src/main/scala/io/iohk/ethereum/vm/OpCode.scala

```diff
@@ -384,6 +384,32 @@ case object EXTCODECOPY extends OpCode(0x3c, 4, 0, G_extcode) {
   }
 }
 
+case object BLOCKHASH extends OpCode(0x40, 1, 1, G_blockhash) with ConstGas {
+  protected def exec[W <: WorldStateProxy[W, S], S <: Storage[S]](state: ProgramState[W, S]): ProgramState[W, S] = {
+    val (blockNumber, stack1) = state.stack.pop
+
+    val hash = if (state.context.env.blockHeader.number - blockNumber > 256 || blockNumber > state.context.env.blockHeader.number) {
```

Comment:
actually definition form yellow paper says:
```
the hash of a block of a particular number, up to a maximum age.
0 is left on the stack if the looked for block number is greater than the current block number or more than 256 blocks behind the current block.
```
so in the case of block verification, it is possible to ask for the hash of current block but it would lead to inconsistency in block mining and block verification.
Good catch, I will change it.